autopilot/contractor.go

```diff
@@ -534,14 +537,17 @@ func (c *contractor) candidateHosts(cfg Config, wanted int) ([]consensus.PublicK
 		return nil, err
 	}
 
-	// filter hosts
+	// fetch low score threshold
+	lowScoreThreshold := float64(0) // TODO
+
+	// filter only usable hosts
 	hosts = hosts[:0]
 	for _, h := range hosts {
 		if used[h.PublicKey.String()] {
 			continue
 		}
 
-		usable, _ := isUsableHost(cfg, ipFilter, Host{h})
+		usable, _ := isUsableHost(cfg, ipFilter, Host{h}, lowScoreThreshold)
```

Comment:
I think it might be problematic to try to define a `lowScoreThreshold`. The score is a combination of multiple weights, so the final number feels kinda...arbitrary? But within the *individual* weights, there are situations where you'd want to mark a host as unusable, e.g. if its version is too low or its prices are extremely high. So I think the process we want is:
1. Filter out hosts that are definitely unusable (blacklisted, unknown settings, etc. -- anything that would result in a score of 0)
2. Score + select from the remaining hosts (none of the scores should be 0!)

Reply:
I've updated the host filter to filter out hosts that are definitely unusable, I actually just removed the settings score because that was binary anyway and moved it into the filter as `hasBadSettings`. So 1. should be covered, then I score the remaining hosts and filter out the 0 scores, there shouldn't be any at this point so I'm logging this event.
Sounds good? If so we can merge this and then I'll probably add some testing to cover the host filter at least a little.